python-checks/src/main/java/org/sonar/python/checks/UndeclaredNameUsageCheck.java

```diff
@@ -120,12 +116,20 @@ private void addNameIssues(Map<String, List<Name>> nameIssues) {
       PreciseIssue issue = subscriptionContext.addIssue(first, first.name() + " is not defined. Change its name or define it before using it");
       list.stream().skip(1).forEach(n -> issue.secondary(n, null));
     });
-    nameIssues.clear();
   }
 
-  private static class ExceptionVisitor extends BaseTreeVisitor {
+  private static class UnresolvedSymbolsVisitor extends BaseTreeVisitor {
+
     private boolean hasWildcardImport = false;
     private boolean callGlobalsOrLocals = false;
+    private Map<String, List<Name>> nameIssues = new HashMap<>();
+
+    @Override
+    public void visitName(Name name) {
+      if (!callGlobalsOrLocals && !hasWildcardImport && name.isVariable() && name.symbol() == null) {
```

Comment:
I think you can now drop `!callGlobalsOrLocals && !hasWildcardImport` on this line.